woocommerce/woocommerce-ios
# [Deposit summary] Analytics continued

Closes: #11234 


## Description


This PR adds the remaining Analytics:
- PAYMENTS_HUB_DEPOSIT_SUMMARY_CURRENCY_SELECTED
- PAYMENTS_HUB_DEPOSIT_SUMMARY_LEARN_MORE_CLICKED
- PAYMENTS_HUB_DEPOSIT_SUMMARY_EXPANDED

Additionally, it moves `PAYMENTS_HUB_DEPOSIT_SUMMARY_SHOWN` to the DepositSummaryView, so it's only logged when that actually appears (not when it's inited.)

Finally, we only fetch the deposit details for WooPayments stores now, so the PAYMENTS_HUB_DEPOSIT_SUMMARY_ERROR event won't be tracked on non-WooPayments stores any more.

## Testing instructions


1. Launch the app and switch to a WooPayments store
2. Navigate to `Menu > Payments`
3. Observe that when the Deposit summary appears, `PAYMENTS_HUB_DEPOSIT_SUMMARY_SHOWN` is tracked with the appropriate number of currencies
4. Tap the summary to expand it
5. Observe that `PAYMENTS_HUB_DEPOSIT_SUMMARY_EXPANDED` is tracked
6. Tap `Learn More about when you'll receive your funds`
7. Observe that `PAYMENTS_HUB_DEPOSIT_SUMMARY_LEARN_MORE_CLICKED` is tracked
8. Close the webview and switch to a different currency in the view
9. Observe that `PAYMENTS_HUB_DEPOSIT_SUMMARY_CURRENCY_SELECTED` is tracked, along with the `currency` property.

### Non-WooPayments stores
1. Switch to a Stripe store
2. Repeat the above test
3. Observe that the Deposit Summary is not shown, and `PAYMENTS_HUB_DEPOSIT_SUMMARY_ERROR` is not tracked.

## Screenshots


<img width="1245" alt="Console logs showing the above analytic events" src="https://github.com/woocommerce/woocommerce-ios/assets/2472348/34cb9c5c-6d1d-4213-a931-c8d50724f628">

---
